_source/_posts/2019-05-07-nodejs-typescript-api.md

```diff
@@ -732,30 +739,26 @@ ReactDOM.render(
 );
 ```
 
-You can create a new React hook to help with authentication as well. This will require you to pass in an `auth` variable, which will then be used to determine whether or not a user is authenticated, find out info about the user, and get the access token. These are then passed back into your React component for user later. Create a new file `src/client/auth.ts`:
+You can create a new React hook to help with authentication as well. You can use the Okta React SDK's `useOktaAuth()` hook to determine whether or not a user is authenticated, find out info about the user, and get the access token. These are then passed back into your React component for user later. Create a new file `src/client/auth.ts`:
 
 ```typescript
 import { useEffect, useState } from "react";
+import { useOktaAuth } from "@okta/okta-react";
 
-export const useAuth = (auth) => {
+export const useAuth = () => {
+  const { authService, authState } = useOktaAuth();
   const [authenticated, setAuthenticated] = useState(null);
   const [user, setUser] = useState(null);
   const [token, setToken] = useState(null);
 
   useEffect(() => {
-    auth.isAuthenticated().then((isAuthenticated) => {
-      if (isAuthenticated !== authenticated) {
-        setAuthenticated(isAuthenticated);
-      }
-    });
+    setAuthenticated(authState.isAuthenticated)
```

Comment:
Since authState.isAuthenticated is synchronous, do you have a need for a separate `authenticated` state?

Reply:
Right, I could condense it to something like the following:

```
export const useAuth = () => {
  const { authService, authState } = useOktaAuth();
  const [user, setUser] = useState(null);
  const [token, setToken] = useState(null);

  useEffect(() => {
    if (authState.isAuthenticated) {
      authService.getUser().then(setUser);
      setToken(`Bearer ${authState.accessToken}`);
    } else {
      setUser(null);
      setToken(null);
    }
  });

  return [user, token];
};
```

Reply:
Changing it to the above seems to cause the authentication to always be false. It could be something else though. I'll revisit once I've fixed the issue below.

Reply:
This would have the same problem that user isn't set.  You can return a function that returns an enclosed value, or....um....